builder/job.go

```diff
@@ -142,6 +143,16 @@ func Build(d *daemon.Daemon, buildConfig *Config) error {
 
 	sf := streamformatter.NewStreamFormatter(buildConfig.JSONFormat)
 
+	var defaultArchiver *archive.Archiver
+	uidMaps, gidMaps := d.GetUidGidMaps()
+	if uidMaps != nil || gidMaps != nil {
```

Comment:
I don't have strong feelings about it, but reading code, when I see `A != nil || B != nil` there is an implicit assumption that the right portion of the `||` is a possibility that we are effectively testing for.